LogMonitor/src/LogMonitor/SystemInfo.cpp

```diff
@@ -0,0 +1,172 @@
+            REG_KEY_DW_DEFAULT_VALUE);
+        Utility::GetDWORDRegKey(
+            hKey,
+            CUR_MAJOR_VER_NUM_STR_VALUE_NAME,
+            mRegistryCurrentVersion.CurrentMajorVersionNumber,
+            REG_KEY_DW_DEFAULT_VALUE);
+        Utility::GetStringRegKey(
+            hKey,
+            PRODUCT_NAME_STR_VALUE_NAME,
+            mRegistryCurrentVersion.ProductName,
+            REG_KEY_STR_DEFAULT_VALUE);
+        break;
+    case ERROR_FILE_NOT_FOUND:
+        logWriter.TraceError(
+            Utility::FormatString(L"Key not found").c_str());
+        break;
+    default:
+        logWriter.TraceError(
+            Utility::FormatString(L"Error opening key").c_str());
+        break;
+    }
+
+    TCHAR buffer[256] = TEXT("");
+    DWORD dwSize = _countof(buffer);
+    int cnf = 0;
+    for (cnf = 0; cnf < ComputerNameMax; cnf++) {
+        if (!GetComputerNameEx((COMPUTER_NAME_FORMAT)cnf, buffer, &dwSize)) {
+            logWriter.TraceError(
+                Utility::FormatString(L"GetComputerNameEx failed %lu", GetLastError()).c_str());
+            return;
+        } else {
+            switch (cnf) {
+            case 0: mCompName.NetBIOS = buffer;
+                break;
+            case 1: mCompName.DnsHostname = buffer;
+                 break;
+            case 2: mCompName.DnsDomain = buffer;
+                 break;
+            case 3: mCompName.DnsFullyQualified = buffer;
+                 break;
+            case 4: mCompName.PhysicalNetBIOS = buffer;
+                 break;
+            case 5: mCompName.PhysicalDnsHostname = buffer;
+                 break;
+            case 6: mCompName.PhysicalDnsDomain = buffer;
+                 break;
+            case 7: mCompName.PhysicalDnsFullyQualified = buffer;
+                 break;
+            default:
+                break;
+            }
+        }
+        dwSize = _countof(buffer);
+        ZeroMemory(buffer, dwSize);
+    }
+
+    SYSTEM_INFO siSysInfo;
+    GetSystemInfo(&siSysInfo);
+    mHardwareInfo.dwOemId = siSysInfo.dwOemId;
+    mHardwareInfo.dwNumberOfProcessors = siSysInfo.dwNumberOfProcessors;
+    mHardwareInfo.dwPageSize = siSysInfo.dwPageSize;
+    mHardwareInfo.dwProcessorType = siSysInfo.dwProcessorType;
+    mHardwareInfo.lpMaximumApplicationAddress = siSysInfo.lpMaximumApplicationAddress;
+    mHardwareInfo.lpMinimumApplicationAddress = siSysInfo.lpMinimumApplicationAddress;
+    mHardwareInfo.wProcessorArchitecture = siSysInfo.wProcessorArchitecture;
+
+
+    LPTSTR lpszVariable;
+    LPTCH lpvEnv;
+    lpvEnv = GetEnvironmentStrings();
+
+    if (lpvEnv == NULL) {
+        logWriter.TraceError(
+            Utility::FormatString(L"GetEnvironmentStrings failed %lu", GetLastError()).c_str());
+    }
+
+    lpszVariable = (LPTSTR) lpvEnv;
+
+    while (*lpszVariable) {
+        lpszVariable += lstrlen(lpszVariable) + 1;
+        if (wcscmp(lpszVariable, L"LOGMONITOR_TELEMETRY=0") == 0) {
```

Comment:
nit: I think there's a better way of reading environment variables, check https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getenvironmentvariable

Reply:
According to docs I'm sharing below there are three options of doing this but your suggestion would be the best to work with here.

The one I had is also in the suggestions. Example three.

[](https://learn.microsoft.com/en-us/windows/win32/procthread/changing-environment-variables )

Reply:
I change this to use the second option!